rotkehlchen/chain/ethereum/uniswap/typing.py

```diff
@@ -0,0 +1,72 @@
+from dataclasses import dataclass
+from typing import (
+    DefaultDict,
+    Dict,
+    List,
+    NamedTuple,
+    Set,
+    Union,
+)
+
+from rotkehlchen.accounting.structures import Balance
+from rotkehlchen.assets.asset import EthereumToken
+from rotkehlchen.assets.unknown_asset import UnknownEthereumToken
+from rotkehlchen.constants import ZERO
+from rotkehlchen.fval import FVal
+from rotkehlchen.typing import (
+    ChecksumEthAddress,
+    Price,
+)
+
+
+class UniswapV2LPToken(NamedTuple):
+    name: str
+    symbol: str
+    decimals: int
+
+
+uniswap_v2_lp_token = (
+    UniswapV2LPToken(name='Uniswap V2', symbol='UNI-V2', decimals=18))
+
+
+@dataclass(init=True, repr=True)
+class LiquidityPoolAsset:
+    asset: Union[EthereumToken, UnknownEthereumToken]
+    balance: FVal
```

Comment:
For this task I have not added next to the field the corresponding subgraph field, sorry.

Explanations below:
```python
class LiquidityPoolAsset:
    asset: Union[EthereumToken, UnknownEthereumToken]
    balance: FVal  # underlying token pool balance, in Uniswap lingo it is called "reserve"
    # ("reserve0" for "token0" and "reserve1" for "token1"
    user_balance: Balance
    asset_usd: Price = Price(ZERO)  # underlying token USD value, requested via Inquirer or subgraph
```

```python
class LiquidityPool:
    asset: UnknownEthereumToken
    assets: List[LiquidityPoolAsset]
    total_supply: FVal
    user_balance: Balance  # address balance of UNI-V2, and its USD value
```

Reply:
Hmm then I am still not sure I follow.

A `LiquidityPool` has many `LiquidityPoolAsset`s. Each `LiquidityPoolAsset` has a `balance` which is the underlying token balance of the asset? But then `user_balance` is the actual LP token balance? If so isn't that duplicating information if `user_balance` of the LP token is duplicated on all underlying `LiquidityPoolAsset` object?

Did I understand something wrong?

Reply:
Hey, is it clearer? Please, let me know what would you keep, discard, rename, etc.

```python
"0xfeF0E7635281eF8E3B705e9C5B86e1d3B0eAb397": [
    {
        "asset": "UNI-V2",  # TODO: swap it with "ethereum_address" and return the checksum address
        "assets": [
            {
                "asset": "SHUF",  # TODO: return a structure for this unknwon asset (incl. ethereum_address, symbol, and name?)
                "asset_usd": "0.2228541114996421832488059018432577",  # USD price per SHUF
                "balance": "129534.791209588069747607",  # Amount of SHUFs in the pool
                "user_balance": {
                    "amount": "2378.268393643042982124462158",  # Amount of SHUFs the user has in the pool (estimated)
                    "usd_value": "530.0068897730016076675764364"  # The USD value of the above
                }
            },
            {
                "asset": "WETH",
                "asset_usd": "381.43",
                "balance": "75.861998428136500877",
                "user_balance": {
                    "amount": "1.392831929209769536731061261",
                    "usd_value": "531.2678827584823943953286968"
                }
            }
        ],
        "total_supply": "2885.285063463439133959",  # Amount of UNI-V2 in the pool
        "user_balance": {
            "amount": "52.974048199782328795",  # Amount of UNI-V2 the user has in the pool
            "usd_value": "1061.274772531484002062905133"  # The USD value of the above
        }
    }
]
```

Reply:
Aha I see. So wait a bit. Perhaps you can use a different naming then? If what you now call `balance` is actually the amount of the underlying asset in the pool just name it properly.

So say "total_amount_in_pool". 

Also rename `asset_usd` to `usd_price` so that it is clear